src/Domain/Creators/TimeDependence/Shape.hpp

```diff
@@ -0,0 +1,184 @@
+        "The outer radius of the Shape map."};
+  };
+
+  using GridToInertialMap =
+        detail::generate_coordinate_map_t<Frame::Grid, Frame::Inertial,
+                                          tmpl::list<ShapeMap>>;
+
+  using options = tmpl::list<InitialTime, LMax, Mass, Spin, Center, InnerRadius,
+                             OuterRadius>;
+
+  static constexpr Options::String help = {
+      "Creates a Shape that conforms to a Kerr horizon of given mass and "
+      "spin."};
+
+  Shape() = default;
+  ~Shape() override = default;
+  Shape(const Shape&) = delete;
+  Shape(Shape&&) = default;
+  Shape& operator=(const Shape&) = delete;
+  Shape& operator=(Shape&&) = default;
+
+  Shape(double initial_time, size_t l_max, double mass,
+        std::array<double, 3> spin, std::array<double, 3> center,
+        double inner_radius, double outer_radius,
+        const Options::Context& context = {});
+
+  auto get_clone() const -> std::unique_ptr<TimeDependence<mesh_dim>> override;
+
+  auto block_maps_grid_to_inertial(size_t number_of_blocks) const
+      -> std::vector<std::unique_ptr<domain::CoordinateMapBase<
+          Frame::Grid, Frame::Inertial, mesh_dim>>> override;
+
+  auto block_maps_grid_to_distorted(size_t number_of_blocks) const
+      -> std::vector<std::unique_ptr<domain::CoordinateMapBase<
+          Frame::Grid, Frame::Distorted, mesh_dim>>> override {
+    using ptr_type =
+        domain::CoordinateMapBase<Frame::Grid, Frame::Distorted, mesh_dim>;
+    return std::vector<std::unique_ptr<ptr_type>>(number_of_blocks);
+  }
+
+  auto block_maps_distorted_to_inertial(size_t number_of_blocks) const
+      -> std::vector<std::unique_ptr<domain::CoordinateMapBase<
+          Frame::Distorted, Frame::Inertial, mesh_dim>>> override {
+    using ptr_type =
+        domain::CoordinateMapBase<Frame::Distorted, Frame::Inertial, mesh_dim>;
+    return std::vector<std::unique_ptr<ptr_type>>(number_of_blocks);
+  }
+
+  auto functions_of_time(const std::unordered_map<std::string, double>&
+                             initial_expiration_times = {}) const
+      -> std::unordered_map<
+          std::string,
+          std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>> override;
+
+ private:
+  // NOLINTNEXTLINE(readability-redundant-declaration)
+  friend bool operator==(const Shape& lhs,
+                         const Shape& rhs);
+
+  using TransitionFunction = domain::CoordinateMaps::
+      ShapeMapTransitionFunctions::ShapeMapTransitionFunction;
+
+  GridToInertialMap grid_to_inertial_map() const;
+  double initial_time_{std::numeric_limits<double>::signaling_NaN()};
+  size_t l_max_{2};
+  double mass_{std::numeric_limits<double>::signaling_NaN()};
+  std::array<double, 3> spin_{
+      make_array<3>(std::numeric_limits<double>::signaling_NaN())};
+  std::array<double, 3> center_{
+      make_array<3>(std::numeric_limits<double>::signaling_NaN())};
+  inline static const std::string function_of_time_name_{"Shape"};
```

Comment:
This assumes we will only have 1 shape map in the domain. Is this an ok assumption? Meaning will we ever want to be able to add two shape time dependences at a time? My guess is no, but just thought I'd ask.

Reply:
After discussion with @nilsdeppe I confirmed that Domains that require the equivalent of two shape maps are highly specialized, so there will be no reason to expect them to make use of the Shape TimeDependence functionality being added, so it's ok to keep this as just "Shape."

Reply:
Ok! Sounds good!